07_dijkstras_algorithm/c++11/01_dijkstras_algorithm.cpp

```diff
@@ -0,0 +1,81 @@
+#include <iostream>
+#include <unordered_map>
+#include <string>
+#include <vector>
+#include <limits>
+
+int main(void)
+{
+  using node_t = std::string;
+  using node_cost_t = std::unordered_map<node_t, unsigned>;
+  using graph_t = std::unordered_map<node_t, node_cost_t>;
+  using parent_graph_t = std::unordered_map<node_t, node_t>;
+
+  // Setup graph
+  graph_t graph;
+  graph.reserve(4U);
+  graph.emplace("start", node_cost_t{{"a", 6}, {"b", 2}});
+  graph.emplace("a", node_cost_t{{"finish", 1}});
+  graph.emplace("b", node_cost_t{{"a", 3},{"finish", 5}});
+  graph.emplace("finish", node_cost_t{});
+  // Setup costs table
+  node_cost_t costs{{"a", 6},{"b", 2},{"finish", std::numeric_limits<unsigned>::max()}};
+  // Setup parents table
+  parent_graph_t parents{{"a", "start"}, {"b", "start"}};
+
+  // A vector of processed nodes
+  std::vector<node_t> processed;
+  processed.reserve(3U);
+  // A lambda function to find the lowest cost node
```

Comment:
find_lowest_cost_node is defined as a lambda function to:
1. Keep the interface the same as the one in the book and other languages.
2. Capture the local variable `processed`.